cli/src/commands/diff.rs

```diff
@@ -0,0 +1,496 @@
+use clap::Parser as ClapParser;
+use codec::Decode;
+
+use frame_metadata::RuntimeMetadataPrefixed;
+use std::collections::HashMap;
+
+use crate::utils::FileOrUrl;
+use color_eyre::owo_colors::OwoColorize;
+
+use scale_info::form::PortableForm;
+use scale_info::Variant;
+
+use subxt_metadata::{
+    ConstantMetadata, Metadata, PalletMetadata, RuntimeApiMetadata, StorageEntryMetadata,
+    StorageEntryType,
+};
+
+/// Explore the differences between two nodes
+///
+/// # Example
+/// ```
+/// subxt diff ./artifacts/polkadot_metadata_small.scale ./artifacts/polkadot_metadata_tiny.scale
+/// subxt diff ./artifacts/polkadot_metadata_small.scale wss://rpc.polkadot.io:443
+/// ```
+#[derive(Debug, ClapParser)]
+pub struct Opts {
+    entry1: FileOrUrl,
+    entry2: FileOrUrl,
```

Comment:
Perhaps a slightly more descriptive name for these like `metadata_or_url1` and `metadata_or_url2` would make the output a little clearer.

I wonder whether adding comments above these would lead to nicer help being shown in the output too :)